ansible/09_create_common-config.yaml

```diff
@@ -1,165 +1,131 @@
-    environment:
-      PATH: "/usr/local/bin:{{ ansible_env.PATH }}"
-      KUBECONFIG: "{{base_path}}/dev-scripts/ocp/{{ ocp_cluster_name }}/auth/kubeconfig"
-
-  - name: copy openstackclient-admin-secret.yaml to {{ base_path }}
-    copy:
-      src: files/ocp/openstackclient-admin-secret.yaml
-      dest: "{{ base_path }}/openstackclient-admin-secret.yaml"
-      owner: ocp
-
-  - name: create openstackclient admin user secret
-    shell: >
-      oc apply -f {{ base_path }}/openstackclient-admin-secret.yaml
-    args:
-      chdir: "{{ base_path }}"
-    environment:
-      PATH: "/usr/local/bin:{{ ansible_env.PATH }}"
-      KUBECONFIG: "{{base_path}}/dev-scripts/ocp/{{ ocp_cluster_name }}/auth/kubeconfig"
-
-  - name: get list of controller internal_api_hostname
-    set_fact:
-      osp_controller_internal_api_hostnames: "{{ osp_controller_internal_api_hostnames|default([]) }} + [ '{{ item.value.internal_api_hostname }}' ]"
-    loop: "{{ osp_inventory.Controller.hosts|dict2items }}"
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
 
   - name: create service secret
     block:
+    - set_fact:
+        osp_worker_yaml_dir: "{{ working_yamls_dir }}/osp-worker"
+
+    - debug:
+        msg: "yamls will be written to {{ osp_worker_yaml_dir }} locally"
+
+    - name: Create yaml dir
+      file:
+        path: "{{ osp_worker_yaml_dir }}"
+        state: directory
+
     - name: delete osp-secrets secret (reinstall/rerun)
       command: oc delete secret osp-secrets -n openstack
-      args:
-        chdir: "{{ base_path }}"
       environment:
-        PATH: "/usr/local/bin:{{ ansible_env.PATH }}"
-        KUBECONFIG: "{{base_path}}/dev-scripts/ocp/{{ ocp_cluster_name }}/auth/kubeconfig"
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
       ignore_errors: true
 
-    - name: create rabbit_transport_url secret
-      block:
-      - set_fact:
-          rabbit_transport_url: "{{ rabbit_transport_url }}{{ (rabbit_transport_url|length > 0) | ternary(',', '', omit) }}guest:{{ osp_global_config_settings.oslo_messaging_rpc_password }}@{{ item }}:5672"
-        with_items:
-        - "{{ osp_controller_internal_api_hostnames }}"
-      - set_fact:
-          rabbit_transport_url: "rabbit://{{ rabbit_transport_url }}/?ssl=0"
-
     - set_fact:
-        rabbit_transport_url: "{{ rabbit_transport_url | b64encode }}"
-        cinder_password: "{{ osp_controller.service_configs['cinder::keystone::auth::password'] | b64encode }}"
-        nova_password: "{{ osp_controller.service_configs['nova::keystone::auth::password'] | b64encode }}"
-        neutron_password: "{{ osp_controller.service_configs['neutron::keystone::auth::password'] | b64encode }}"
-        placement_password: "{{ osp_controller.service_configs['placement::keystone::auth::password'] | b64encode }}"
+        rabbit_transport_url: "{{ 'rabbit://cell1:passw0rd@rabbitmq.openstack.svc:5672/cell1?ssl=0' | b64encode }}"
```

Comment:
FYI, I saw in a @dprince patch that you can specify 'stringData' (iirc) in the Secret yaml, and it will encode it for you.